server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -57,11 +60,13 @@
 
     public static class Names {
         public static final String IGNORE_MALFORMED = "ignore_malformed";
+        public static final ParseField ACCEPT_Z_VALUE = new ParseField("accept_z_value");
     }
 
     public static class Defaults {
         public static final Explicit<Boolean> IGNORE_MALFORMED = new Explicit<>(false, false);
         public static final GeoPointFieldType FIELD_TYPE = new GeoPointFieldType();
+        public static final Explicit<Boolean> ACCEPT_Z_VALUE = new Explicit<>(true, false);
```

Comment:
true by default? shouldn't it be false like the docs say? (we might need some logic to make it true for pre-6.3 indices in case we used to accept it in the past)